forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -372,15 +372,24 @@ async fn validate(
         let db_path = tmp_db_path.path().join(&config.chain.name);
         let db = open_db(&db_path, config.db_config())?;
 
-        let chain_store = Arc::new(ChainStore::new(db, config.chain.clone()));
-
         let genesis = read_genesis_header(
             config.client.genesis_file.as_ref(),
             config.chain.genesis_bytes(),
-            &chain_store,
+            &db,
         )
         .await?;
 
+        let genesis_ts = Tipset::try_from(&genesis.clone())?;
+
+        let chain_store = Arc::new(ChainStore::new(
+            db,
+            config.chain.clone(),
+            Arc::new(genesis_ts.clone()),
+        )?);
+
+        chain_store.set_genesis(&genesis.clone())?;
+        chain_store.set_heaviest_tipset(Arc::new(genesis_ts.clone()))?;
```

Comment:
Do we need to set the genesis and heaviest tipset here or has it already been done in `ChainStore::new`?